pkg/controllers/dynakube/extension/secret.go

```diff
@@ -98,3 +75,82 @@ func (r *reconciler) buildSecret(eecToken dttoken.Token, otelcToken dttoken.Toke
 func (r *reconciler) getSecretName() string {
 	return r.dk.Extensions().GetTokenSecretName()
 }
+
+func (r *reconciler) newSecret() (*corev1.Secret, error) {
+	newEecToken, err := dttoken.New(eecConsts.TokenSecretValuePrefix)
+	if err != nil {
+		log.Info("failed to generate eec token")
+		conditions.SetSecretGenFailed(r.dk.Conditions(), secretConditionType, errors.Wrap(err, "error generating eec token"))
+
+		return nil, err
+	}
+
+	newOtelcToken, err := dttoken.New(consts.DatasourceTokenSecretValuePrefix)
+	if err != nil {
+		log.Info("failed to generate otelc token")
+		conditions.SetSecretGenFailed(r.dk.Conditions(), secretConditionType, errors.Wrap(err, "error generating otelc token"))
+
+		return nil, err
+	}
+
+	newSecret, err := r.buildSecret([]byte(newEecToken.String()), []byte(newOtelcToken.String()))
+	if err != nil {
+		log.Info("failed to generate extension secret")
+		conditions.SetSecretGenFailed(r.dk.Conditions(), secretConditionType, err)
+
+		return nil, err
+	}
+
+	return newSecret, nil
+}
+
+func (r *reconciler) migrateSecret(existingSecret *corev1.Secret) (*corev1.Secret, error) {
+	const oTelcTokenKey = "otelc.token"
+
+	if _, exists := existingSecret.Data[consts.DatasourceTokenSecretKey]; exists {
+		// no migration needed, datasource token file name already correct
+		return nil, ErrNoMigration
```

Comment:
i don't think logically returnig error for case when migration is no needed is good idea, instead change function to:
```
migrateSecret(target *corev1.Secret) (*corev1.Secret, bool) {
isMigrated := false

...
return secret, isMigrated
```